### ISSUE TITLE
comparison of Gem::Version with String failed?

### ISSUE THREAD

**Issue by kitplummer** [jamtur01/puppet-hipchat#24]
Error: Could not send report: Error 400 on SERVER: Could not autoload puppet/reports/hipchat: comparison of Gem::Version with String failed

Didn't really dig, but figured you'd want to see any issues found.

**Comment by mvisonneau** [comment]
Thanks for the workaround, I have the same problem :+1:

**Comment by james-powis** [comment]
My clash with version was slightly different:
`Error: Could not send report: Error 400 on SERVER: Could not autoload puppet/reports/hipchat: undefined method `version' for nil:NilClass`

from the puppetserver log (viewable only on first agent report only):

```
2016-03-14 14:57:11,075 ERROR [qtp1541639922-64] [puppet-server] Puppet Could not autoload puppet/reports/hipchat: undefined method `version' for nil:NilClass
/etc/puppetlabs/code/environments/puppet_masters/modules/puppet_hipchat/lib/puppet/reports/hipchat.rb:44:in `(root)'
org/jruby/RubyModule.java:2344:in `module_eval'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/classgen.rb:136:in `genthing'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/classgen.rb:59:in `genmodule'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/reports.rb:57:in `register_report'
/etc/puppetlabs/code/environments/puppet_masters/modules/puppet_hipchat/lib/puppet/reports/hipchat.rb:10:in `(root)'
org/jruby/RubyKernel.java:1091:in `load'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/autoload.rb:1:in `(root)'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/autoload.rb:59:in `load_file'
...
```
The issue was traced down to the hipchat gem being missing in my puppetserver gem path resolved with (note the --http-proxy is required if your puppet.conf does not have proxy set):
`/opt/puppetlabs/bin/puppetserver gem install --http-proxy http://proxyserver:80 hipchat`

Under various versions of puppetserver the puppetserver intermittently has included the puppet agents ruby lib path. This can cause unexpected behavior as I believe was being reported here.

A quick and dirty fix on puppetversion > 4.0.0 would be to install puppetserver_gem provider and invoke the puppet_hipchat module via:
```
class { 'puppet_hipchat':
  ...
  install_hc_gem => true,
  provider => 'puppetserver_gem'
  ...
}
```

But it is worth noting your puppetserver must be entirely aware of the http_proxy as well...

**Comment by james-powis** [comment]
closing issue due to inactivity.
